dot/sync/chain_sync.go

```diff
@@ -1012,11 +1018,12 @@ func doResponseGrowsTheChain(response, ongoingChain []*types.BlockData, startAtB
 	firstBlockInResponse := response[0]
 	firstBlockExactIndex := firstBlockInResponse.Header.Number - startAtBlock
 	if firstBlockExactIndex != 0 {
-		leftElement := ongoingChain[firstBlockExactIndex-1]
-		if leftElement != nil && !compareParentHash(leftElement, firstBlockInResponse) {
-			return false
+		if firstBlockExactIndex < uint(expectedTotal) {
```

Comment:
You can leave the conditions in a single if 
```suggestion
		if firstBlockExactIndex != 0 && firstBlockExactIndex < uint(expectedTotal) {
```